packages/core/src/utilities/derivedImageMetadataProvider.ts

```diff
@@ -0,0 +1,41 @@
+import { metaData } from '..';
+
+const state: Record<string, any> = {}; // Calibrated pixel spacing per imageId
+/**
+ * Simple metadata provider that stored the derived metadata in memory.
+ */
+const metadataProvider = {
+  /**
+   * Adds metadata for an imageId.
+   * @param imageId - the imageId for the metadata to store
+   * @param payload - the payload composed of new calibrated pixel spacings
+   */
+  add: (imageId: string, payload: any): void => {
+    const type = payload.type;
+
+    if (!state[imageId]) {
+      state[imageId] = {};
+    }
+
+    state[imageId][type] = {
+      ...payload.metadata,
```

Comment:
What about an object clone?  The problem with this is that you don't clone arrays and nested objects, so it still shares some data with the original item.